packages/core/src/services/scales-cartesian.ts

```diff
@@ -319,12 +403,15 @@ export class CartesianScales extends Service {
 	protected findMainHorizontalAxisPosition() {
 		const options = this.model.getOptions();
 		const axesOptions = Tools.getProperty(options, "axes");
+		const dualAxes = this.isDualAxes();
 
 		// If top axis has been specified as `main`
 		if (
 			(Tools.getProperty(axesOptions, AxisPositions.BOTTOM) === null &&
 				Tools.getProperty(axesOptions, AxisPositions.TOP) !== null) ||
-			Tools.getProperty(axesOptions, AxisPositions.TOP, "main") === true
+			Tools.getProperty(axesOptions, AxisPositions.TOP, "main") === true ||
+			dualAxes && Tools.getProperty(axesOptions, AxisPositions.BOTTOM, "datasets")
```

Comment:
couple of points here:
1. `dualAxes && Tools.getProperty(axesOptions, AxisPositions.BOTTOM, "datasets")` should be in brackets
2. What if `datasets` exists & is an empty array?